### DISCUSSION TITLE
Var or implicit type?

### DISCUSSION THREAD

**Discussion by GeorgeLeithead** [CommunityToolkit/Maui#507] [General]
Hi all,

I personally prefer using implicit types.  However, whislt contributing and receiving contributions from other (great) contributors, I noticed that SOME change the code to use `var` rather than **implicit types**.  Should the [CONTRIBUTING.md](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md) specify var or implicit types, and could this be enforced by `.editorconfig` or `StyleCop.Analyzers'?

**Comment by GeorgeLeithead** [comment]
Any thoughts, opinions, ideas?

**Comment by VladislavAntonyuk** [comment]
I would recommend something like this:
```csharp
var myClass = new MyClass(); // use var if the type is clear

MyClass myClass = new(); // alternative variant

var myString = GetMyString(); // type is clear -> use var

MyClass myClass = GetCalculationResult(); // type not clear -> do not use var

**Comment by CliffAgius** [comment]
Closed as answered
